### PR TITLE
Add OFFLINE mode to QEMU backend

### PR DESCRIPTION
**Pull request by asdil12**
This mode will disable all networkinterfaces for the VM.

https://progress.opensuse.org/issues/33262

Required by https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/4767

### CODE REVIEW

File: backend/qemu.pm [hunk 1]
```diff
@@ -570,6 +570,10 @@ sub start_qemu {
     my $num_networks = 1;
```
Comment (Martchus) [inline backend/qemu.pm]: What about:
```
my $num_networks = max($vars->{OFFLINE} ? 0 : 1, scalar @nicmac, scalar @nicvlan, scalar @tapdev);
```

Comment (coolo) [reply backend/qemu.pm]: not sure this is an improvement

Comment (asdil12) [reply backend/qemu.pm]: For some reason it doesn't seem to work this way.

**Comment by asdil12** [comment]
set to `OFFLINE_SUT`

File: distribution.pm [hunk 2]
```diff
@@ -192,6 +192,12 @@ sub script_output {
     $args{type_command} //= length($script) < 80;
     my $script_path = "/tmp/script$marker.sh";
 
+    # prevent use of network for offline installations
+    if (testapi::get_var 'OFFLINE_SUT') {
+        testapi::record_info('forced type_cmd', "Forced typing in the command as we are offline");
```
Comment (coolo) [inline distribution.pm]: I think typing already includes the 'in' - there is little out typing at least.

File: testapi.pm [hunk 3]
```diff
@@ -1847,19 +1847,24 @@ sub upload_logs {
...
 
-    bmwqemu::log_call(file => $file, %args);
-    my $basename = basename($file);
-    my $upname   = ($args{log_name} || $autotest::current_test->{name}) . '-' . $basename;
-    my $cmd      = "curl --form upload=\@$file --form upname=$upname ";
-    $cmd .= autoinst_url("/uploadlog/$basename");
-    if ($failok) {
-        # just use script_run so we don't care if the upload fails
-        script_run($cmd, $timeout);
+    if (get_var 'OFFLINE_SUT') {
```
Comment (coolo) [inline testapi.pm]: use get_var()

File: testapi.pm [hunk 4]
```diff
@@ -1847,19 +1847,24 @@ sub upload_logs {
...
+        record_info('upload skipped', "Skipped uploading log file '$file' as we are offline");
```
Comment (coolo) [inline testapi.pm]: return here - or refactor to avoid yet another if/else

File: testapi.pm [hunk 5]
```diff
@@ -1891,16 +1896,21 @@ C<$nocheck> parameter:
...
     my ($file, $public, $nocheck) = @_;
 
-    bmwqemu::log_call(file => $file, public => $public, nocheck => $nocheck);
-    my $cmd = "curl --form upload=\@$file ";
-    $cmd .= "--form target=assets_public " if $public;
-    my $basename = basename($file);
-    $cmd .= autoinst_url("/upload_asset/$basename");
-    if ($nocheck) {
-        type_string("$cmd\n");
+    if (get_var 'OFFLINE_SUT') {
```
Comment (coolo) [inline testapi.pm]: same here. no else but return

**Comment by asdil12** [comment]
@coolo updated.

**Pull request merged**